jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java

```diff
@@ -0,0 +1,72 @@
+package org.jsonschema2pojo.util;
+
+import org.jsonschema2pojo.URLProtocol;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+public class URLUtilTest {
+
+    @Test
+    public void testParseProtocol() {
+        Assert.assertEquals(URLProtocol.HTTPS, URLUtil.parseProtocol("https://testhttps.protocol"));
+        Assert.assertEquals(URLProtocol.HTTP, URLUtil.parseProtocol("http://testhttp.protocol"));
+        Assert.assertEquals(URLProtocol.FILE, URLUtil.parseProtocol("file:/testfile/protocol"));
+        Assert.assertEquals(URLProtocol.CLASSPATH, URLUtil.parseProtocol("classpath:/test/classpath/class.java"));
+        Assert.assertEquals(URLProtocol.JAVA, URLUtil.parseProtocol("java:test/java"));
+        Assert.assertEquals(URLProtocol.RESOURCE, URLUtil.parseProtocol("resource:test/resource"));
+        Assert.assertEquals(URLProtocol.NO_PROTOCOL, URLUtil.parseProtocol("there are no protocol: at all"));
+    }
+
+    @Test
+    public void testParseUrlNoProtocol() throws MalformedURLException {
+        String url = "This: not url at all";
+        File file = new File(url);
+        URL expectedURL = file.toURI().toURL();
+        Assert.assertEquals(expectedURL, URLUtil.parseURL(url));
+    }
+
+    @Test
+    public void testParseUrlDefault() throws MalformedURLException {
+        String url = "file:this/is/url";
+        Assert.assertEquals(new URL(url), URLUtil.parseURL(url));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseUrlException() {
```

Comment:
I would rename this to `testParseThrowsExceptionWhenURLIsMalformed`.

Reply:
Yeah, sorry I forgot to naming it before commit :D